inference-engine/src/transformations/src/transformations/common_optimizations/fq_mul_fusion.cpp

```diff
@@ -0,0 +1,98 @@
+//                  v                           v  v  v       v                         v
+//             +----------+                    +----------------------------------------------+
+//             | Multiply | <--- C6            |                 FakeQuantize                 |
+//             +----+-----+                    +----------------------------------------------+
+//                  |                                                   |
+//                  v                                                   v
+//
+
+ngraph::pass::FakeQuantizeMulFusion::FakeQuantizeMulFusion() {
+  const auto fq_output_low_p = ngraph::pattern::wrap_type<opset4::Constant>();
+  const auto fq_output_high_p = ngraph::pattern::wrap_type<opset4::Constant>();
+
+  const auto fq_node_p = ngraph::pattern::wrap_type<opset4::FakeQuantize>(
+      {ngraph::pattern::wrap_type<opset4::Constant>(),
+       ngraph::pattern::wrap_type<opset4::Constant>(),
+       ngraph::pattern::wrap_type<opset4::Constant>(),
+       fq_output_low_p,
+       fq_output_high_p},
+      pattern::consumers_count(1));
+
+  const auto mul_constant_p = ngraph::pattern::wrap_type<opset4::Constant>();
+  const auto mul_node_p = ngraph::pattern::wrap_type<opset4::Multiply>(
+      {fq_node_p, mul_constant_p}, pattern::consumers_count(1));
+
+  ngraph::matcher_pass_callback callback = [=](pattern::Matcher &m) {
+    auto pattern_map = m.get_pattern_value_map();
+
+    const auto fq_node = std::dynamic_pointer_cast<ngraph::opset4::FakeQuantize>(
+        pattern_map[fq_node_p].get_node_shared_ptr());
+    const auto output_low_const = pattern_map[fq_output_low_p].get_node_shared_ptr();
+    const auto output_high_const = pattern_map[fq_output_high_p].get_node_shared_ptr();
+    const auto mul_node = pattern_map[mul_node_p].get_node_shared_ptr();
+    const auto mul_constant = pattern_map[mul_constant_p].get_node_shared_ptr();
+
+    auto fq_data_shape = fq_node->input_value(0).get_partial_shape();
+    const bool mul_constant_matches_fq_data = PartialShape::broadcast_merge_into(
+        fq_data_shape, mul_constant->get_output_partial_shape(0), fq_node->get_auto_broadcast());
+
+    if (!mul_constant_matches_fq_data) {
+      return false;
+    }
+
+    try {
+      // create two copies of the original Mul node and use them to multiply the FQ out_* constants
+      // the following 2 lines might throw a validation error if the mul_constant's shape
+      // does not match the shape of of the output_*_const - in this case we can't modify the graph
+      const auto multiplied_out_low = mul_node->clone_with_new_inputs({output_low_const, mul_constant});
+      const auto multiplied_out_high = mul_node->clone_with_new_inputs({output_high_const, mul_constant});
+
+      // attach the new Mul nodes to the third and fourth input of the FQ node
+      const auto fq_out_low_input = fq_node->input(3);
+      fq_out_low_input.replace_source_output(multiplied_out_low->output(0));
```

Comment:
I suggest to do replacement in this way: 
1. Prepare new inputs with Multiply for FQ (you can keep as you do `mul_node->clone_with_new_inputs` but I suggest to avoid CF after this pass and make folding manually here like we do in lin op fusion: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/transformations/src/transformations/lin_op_sequence_fusion.cpp#L53)
2. Make a FQ node copy with new inputs (`fq->clone_with_new_inputs({fq->input_value(0), fq->input_value(1), fq->input_value(2), multiplied_out_low, multiplied_out_high}`)
3. `replace_node(fq, new_fq);`
4. `new_fq->set_friendly_name(fq->get_friendly_name)`
5. `copy_runtime_info({fq, mul_node}, new_fq)`